ingester/util/ticker.go

```diff
@@ -0,0 +1,57 @@
+package util
+
+import (
+	"time"
+
+	"chainlink/ingester/logger"
+)
+
+// TickerService is an interface that provides
+// basic ticking functionality to a service
+type TickerService interface {
+	Start()
+	Stop()
+	Tick()
+}
+
+// Ticker is the implementation of the TickerService
+// interface, managing the lifecycle of a ticking service
+type Ticker struct {
+	Ticker *time.Ticker
+	Name   string
+	Impl   TickerService
+	Done   chan struct{}
+	Exited chan struct{}
+}
+
+// Start will start the ticker service
+func (t *Ticker) Start() {
+	go t.tick()
+}
+
+// Stop will stop the ticket service, waiting
+// for any any tick to first complete
+func (t *Ticker) Stop() {
+	close(t.Done)
+	logger.Info("Waiting for service to exit")
+	<-t.Exited
+	logger.Info("Safely exited")
+}
+
+func (t *Ticker) tick() {
+	logger.Info("Running initial service tick")
+	t.Impl.Tick()
```

Comment:
Why is there an initial Tick ? Feels like maybe this should be documented at least /cc @jleeh because I think this is from agg-monitor.